### PR TITLE
export modern CMake interface target

### PR DESCRIPTION
**Pull request by dirk-thomas**
Related to ros2/ros2#915.

### CODE REVIEW

File: orocos_kdl/src/CMakeLists.txt [hunk 1]
```diff
@@ -128,6 +131,8 @@ ENDIF()
 
 # Needed so that the generated config.h can be used
 INCLUDE_DIRECTORIES(${CMAKE_CURRENT_BINARY_DIR})
```
Comment (ivanpauno) [inline orocos_kdl/src/CMakeLists.txt]: can this line be replace by modern cmake?

File: orocos_kdl/src/CMakeLists.txt [hunk 2]
```diff
@@ -103,6 +103,9 @@ ENDIF()
 
 ADD_LIBRARY(orocos-kdl ${LIB_TYPE} ${KDL_SRCS} config.h)
 
+TARGET_INCLUDE_DIRECTORIES(orocos-kdl PUBLIC
```
Comment (ivanpauno) [inline orocos_kdl/src/CMakeLists.txt:106]: can this line be combined with L134?

**Review by ivanpauno — APPROVED** [review]
I've left some questions, otherwise LGTM!

Can this be contributed upstream?

File: orocos_kdl/src/CMakeLists.txt (hunk 1 above)
Comment (mjcarroll) [reply orocos_kdl/src/CMakeLists.txt]: I think that's what line 134 is doing?  Maybe this can just be dropped.

Comment (dirk-thomas) [reply orocos_kdl/src/CMakeLists.txt]: Correct, removed in 3efaaf9c97e6ec74cda2bd72ad52382fb9f373d0.

File: orocos_kdl/src/CMakeLists.txt (hunk 2 above)
Comment (dirk-thomas) [reply orocos_kdl/src/CMakeLists.txt:106]: I just tried to maintain the existing structure as much as possible. And both lines exist for different reasons. So I will keep it as is.

**Pull request merged**